packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -339,7 +339,10 @@ export const GridMixin = (superClass) =>
         return 0;
       }
 
-      const columnWidth = Math.max(this.__getIntrinsicWidth(col), this.__getDistributedWidth(col.parentElement, col));
+      const columnWidth = Math.max(
+        this.__getIntrinsicWidth(col),
+        this.__getDistributedWidth((col.assignedSlot || col).parentElement, col),
+      );
```

Comment:
The new tests pass even without this change.

Instead of checking for errors, they could check if the grid renders the cells correctly?

Reply:
Changed tests to check the values instead.